### PR TITLE
feat(orchestrator): workflow editor modal

### PR DESCRIPTION
**Pull request by batzionb**
workflow edit modal in workflows list page and workflow viewer page
image

### CODE REVIEW

**Comment by caponetto** [comment]
@batzionb could you please rebase your branch with `KOGITO-9778-setup`?
Since https://github.com/caponetto/backstage-plugins/pull/58 has been merged, you will have to go to `Settings -> Feature Flags -> Developer mode` to enable authoring (create/edit/delete).

**Comment by caponetto** [comment]
This [patch](https://gist.github.com/caponetto/f5c1566f8e3c1aa90b6f655044d8a1d1) should fix the build.

File: plugins/orchestrator/src/components/WorkflowDialog/WorkflowDialog.tsx [hunk 1]
```diff
@@ -55,9 +60,16 @@ export const WorkflowDialog = (
...
+            ref={forwardedRef}
+          />
         </Box>
       </DialogContent>
+      {props.dialogActions}
     </Dialog>
   );
 };
+
+export const WorkflowDialog = forwardRef(RefForwardingWorkflowDialog);
```
Comment (caponetto) [inline plugins/orchestrator/src/components/WorkflowDialog/WorkflowDialog.tsx:75]: Are you using this ref?

**Review by caponetto — COMMENTED** [review]
My only observation is that the viewer page is not reloading after saving.

https://github.com/caponetto/backstage-plugins/assets/638737/76765241-3596-4a15-8106-a5cd9650aa8c

File: plugins/orchestrator/src/components/WorkflowDialog/WorkflowDialog.tsx (hunk 1 above)
Comment (batzionb) [reply plugins/orchestrator/src/components/WorkflowDialog/WorkflowDialog.tsx:75]: https://github.com/caponetto/backstage-plugins/pull/59/files#diff-39da7eb41123e248a4fd6670e4e3ee395e95dce50cf17a76de47ebbfec551e4fR82

**Comment by batzionb** [comment]
> My only observation is that the viewer page is not reloading after saving.
> 
>  Screen.Recording.2023-12-18.at.09.43.51.mov

If you look at the networking tab, you can see that it is requesting the data again but because of a bug I reported in the backend, the data doesn't return updated
It either returns quickly with the wrong data, or takes time and after about half a minute returns the correct data

See issue https://issues.redhat.com/browse/FLPATH-790

**Comment by batzionb** [comment]
> > My only observation is that the viewer page is not reloading after saving.
> > Screen.Recording.2023-12-18.at.09.43.51.mov
> 
> If you look at the networking tab, you can see that it is requesting the data again but because of a bug I reported in the backend, the data doesn't return updated It either returns quickly with the wrong data, or takes time and after about half a minute returns the correct data
> 
> See issue https://issues.redhat.com/browse/FLPATH-790

**Review by caponetto — APPROVED** [review]

**Pull request merged**